lib/include/openamp/virtio.h

```diff
@@ -138,6 +138,20 @@ struct virtio_device_id {
 #define VIRTIO_TRANSPORT_F_START      28
 #define VIRTIO_TRANSPORT_F_END        32
 
+/*
+ * Read the virtio device configuration member.
+ */
+#define virtio_read_config_member(vdev, structname, member, _ptr) \
+	virtio_read_config((vdev), metal_offset_of(structname, member), \
+			   _ptr, sizeof(*(_ptr)))
+
+/*
+ * Write the virtio device configuration member.
+ */
+#define virtio_write_config_member(vdev, structname, member, _ptr) \
+	virtio_write_config((vdev), metal_offset_of(structname, member), \
+			    _ptr, sizeof(*(_ptr)))
+
```

Comment:
Declaring a local variable in the macro should solve the build issue

```
#define virtio_write_config_member(vdev, structname, member, _ptr)  do { \
	int ptr = (_ptr); \
	virtio_write_config((vdev), metal_offset_of(structname, member), \
			    ptr, sizeof(*(ptr)));
	} while (0)
```

Reply:
Because I need use the sizeof(member), i find a new way, please review again.

Reply:
> Because I need use the sizeof(member), i find a new way, but have same issue :(.

Perhaps this could work:

```
#define virtio_write_config_member(vdev, structname, member, ptr) do { \
    __typeof__(structname) __tmp_structname = (structname); \
    __typeof__(member) __tmp_member = (member); \
    virtio_write_config(vdev, metal_offset_of(__tmp_structname, __tmp_member), \
                        ptr, sizeof(((__tmp_structname *)0)->__tmp_member)); \
} while (0)
```
but is it work for every compiler...?

@glneo  concerns seems valid. This macro introduces an additional API, but it may not offer significant benefits to the library.